pkgs/applications/editors/vim/plugins/nvim-treesitter/grammars.lua

```diff
@@ -0,0 +1,39 @@
+local parsers = require("nvim-treesitter.parsers")
```

Comment:
This script tries to dump parser URL and merge the lockfile. There's no need to introduce Lua as another script language to do this. Notice that,
1. We could dump the config in json with
   `nvim --headless +'lua print(vim.json.encode(require("nvim-treesitter.parsers").get_parser_configs()))' +'quit!'`
2. The lockfile is located at `"${nvim-treesitter}/lockfile.json"`

And then this logic can be merged into the python script below.

Reply:
`vim.json.encode` is nondeterministic and arbitrarily shuffles the items in the table, also `get_parser_configs` includes a lot of information that is not really useful for the update script which can make the json a lot bigger then it needs to be

I didn't know we can use `--headless` with `print` here, I will give it a try

Reply:
> `vim.json.encode` is nondeterministic

You can just run the nvim command in the python script, and then sort keys. Then the intermediate data is not stored.

>  which can make the json a lot bigger then it needs to be

This script is run only once during updating yes? That's not so often.